pkg/kubectl/cmd/testing/fake.go

```diff
@@ -515,7 +516,11 @@ func (f *fakeAPIFactory) Decoder(bool) runtime.Decoder {
 }
 
 func (f *fakeAPIFactory) JSONEncoder() runtime.Encoder {
-	return testapi.Default.Codec()
+	if f.tf.JSONEncoder != nil {
```

Comment:
Try with the unstructuredSerializer. We already have apply tests that persist deployments, which are also in the extensions group, that do not need to do this